src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php

```diff
@@ -124,6 +127,82 @@ protected function json($data, $status = 200, $headers = array(), $context = arr
     }
 
     /**
+     * Returns a BinaryFileResponse object with original or customized file name and disposition header.
+     *
+     * @param File|string $file        File object, path to file or string with content to be sent as response
+     * @param string|null $fileName    File name to be sent to response or null (will use original file name)
+     * @param string      $disposition Disposition of response ("attachment" is default, other type is "inline")
+     *
+     * @return BinaryFileResponse
+     *
+     * @throws \LogicException|\InvalidArgumentException|\RuntimeException
+     */
+    protected function file($file, $fileName = null, $disposition = ResponseHeaderBag::DISPOSITION_ATTACHMENT)
+    {
+        $deleteFileAfterSend = false;
+
+        if (!is_string($file) && !$file instanceof File) {
+            throw new \InvalidArgumentException(sprint('The "%s" method expects first argument to be a string or an instance of "%s"', __METHOD__, File::class));
+        }
+
+        if (is_string($file) && file_exists($file)) {
+            $file = new File($file);
+        }
+
+        if (is_string($file)) {
+            if ('' === $file) {
+                throw new \InvalidArgumentException('File content can\'t be empty.');
+            }
+
+            if (empty($fileName)) {
+                throw new \InvalidArgumentException('File name can\'t be empty.');
+            }
+            $cacheDir = $this->container->get('kernel')->getCacheDir();
+            $tmpName = md5($fileName);
+            $tmpPath = $cacheDir.DIRECTORY_SEPARATOR.$tmpName;
```

Comment:
Using a hash of the file name as the temp file name is a bad idea, if you try to serve 2 different files to 2 different clients with the same name at the same time they will get the same file instead of 2 different ones.